### DISCUSSION TITLE
Alert Mechanism for GitHub Self-Hosted Runners

### DISCUSSION THREAD

**Discussion by arjungoel-kimbal** [community/community#160558] [Actions]
Hi, I have configured a github self-hosted runner on cloud server (Amazon EC2 Windows machine). The issue which I realized while working with self-hosted runners are in case the runner goes offline either the Windows Service gets stopped or deleted and as a user I am not aware of it. 

There is a **_diag** folder which logs the runner state but I still don't have any option where I will receive the email/slack/teams notification in case the runner goes **Offline**. This is one of the common features which we find in most of the tools from monitoring and observability point of view.

Is there a way to accomplish this?

**Comment by VOVELEE** [comment]
I am also interested in this functionality.
It would be nice to receive notification via Webhooks

**Reply by nebuk89** [reply]
If we added this to something like the audit logs at the org level would this be ok?

**Comment by AmilaDevops** [comment]
I also finding a way to accomplish this,  as such I want to notify to Azure insights to send email alerts about these. I think I can use azure ARC enabled to conect with Github Self-hosted runner VM and using a data collection rule to collect Windows-Service Processes logs/events in the GH Self hosted VM. But it more complex. So better if you have that feature on GH Self-hosted runners it self than just log outputs.  Thanks
